.build/run-ci.sh

```diff
@@ -0,0 +1,157 @@
+    -e "/repositoryBranch:/s|:.*$|: \"$REPO_BRANCH\"|" \
+    -e "/repositoryUrl:/s|:.*$|: \"$REPO_URL\"|" jenkins-deployment.yaml
+
+# Add Helm Jenkins Operator repository
+echo "Adding Helm repository for Jenkins Operator..."
+helm repo add --namespace ${KUBE_NS} jenkins https://raw.githubusercontent.com/jenkinsci/kubernetes-operator/master/chart
+
+# Install Jenkins Operator using Helm
+echo "Installing Jenkins Operator..."
+#helm upgrade --namespace ${KUBE_NS} --install jenkins-operator jenkins/jenkins-operator --set jenkins.enabled=false --set jenkins.backup.enabled=false --version 0.8.0-beta.2 
+
+echo "Jenkins Operator installed successfully!" # condition to check if above command was success
+
+# deploy jenkins Instance TODO jenkins file parameter
+kubectl apply --namespace ${KUBE_NS} -f ${CASSANDRA_DIR}/.build/jenkins-deployment.yaml
+
+TOKEN=$(kubectl  get secret jenkins-operator-credentials-example -o jsonpath="{.data.token}" | base64 --decode)
+
+# Trigger a new build and capture the response headers
+response_headers=$(curl -i -X POST http://localhost:8080/job/k8s-e2e/buildWithParameters -u jenkins-operator:$TOKEN --data-urlencode "TEST_STAGES_TO_RUN=$INCLUDE_TEST_STAGE" 2>&1)
+
+
+queue_url=$(echo "$response_headers" | grep -i "Location" | awk -F ": " '{print $2}' | tr -d '\r')
+queue_item_number=$(basename "$queue_url")
+
+# Construct the complete URL to retrieve build information
+queue_json_url="http://localhost:8080/queue/item/$queue_item_number/api/json"
+
+# Wait for the build number to become available (querying the API)
+build_number=""
+while [ -z "$build_number" ] || [ "$build_number" == "null" ]; do
+    build_info=$(curl -s "$queue_json_url" -u jenkins-operator:$TOKEN)
+    build_number=$(echo "$build_info" | jq -r '.executable.number')
+    if [ -z "$build_number" ] || [ "$build_number" == "null" ]; then
+        echo "Build number not available yet. Waiting..."
+        sleep 5  # Adjust the polling interval as needed
+    fi
+done
+
+echo "build_number $build_number"
+BUILD_DIR=/var/lib/jenkins/jobs/k8s-e2e/builds
+POD_NAME=jenkins-example
+LATEST_BUILD=$(kubectl exec -it $POD_NAME -- /bin/bash -c "ls -t $BUILD_DIR | head -n 1" | tr -d '\r')
+
+CONSOLE_LOG_FILE="$BUILD_DIR/$build_number/log"
+
+# Define a function to check if "FINISHED" is in the consoleLog
+check_finished() {
+    kubectl exec -n $KUBE_NS $POD_NAME -- cat "$CONSOLE_LOG_FILE" | grep -q "Finished"
+}
+
+# Continuously check for "FINISHED"
+while true; do
+    if check_finished; then
+        echo "Build has finished."
+        break
+    else
+        echo "Build is still in progress."
+    fi
+    sleep 5  # Adjust the sleep interval as needed
+done
+
+LOCAL_DIR=.
+mkdir -p $LOCAL_DIR/$build_number
+kubectl cp -n $KUBE_NS $POD_NAME:$BUILD_DIR/$build_number/log $LOCAL_DIR/$build_number/log
+kubectl cp -n $KUBE_NS $POD_NAME:$BUILD_DIR/$build_number/junitResult.xml $LOCAL_DIR/$build_number/junitResult.xml
+
+
```

Comment:
we're missing here the teardown of the jenkins stack.
and this needs to be optional via a parameter.

Reply:
you can leave this, i will tackle it.

Reply:
Since for now, the jenkins deployment is working with default namespace only (the limitation we mentioned earlier), so there is no need to delete the default namespace, we can remove the delete command (for namespace)